File: code/nxt-integration/src/main/java/org/fbme/integration/nxt/importer/CompositeFbTypeNxtImporter.java
package org.fbme.integration.nxt.importer;

import org.fbme.lib.iec61499.declarations.CompositeFBTypeDeclaration;
import org.fbme.lib.iec61499.parser.CompositeFBTypeConverter;
import org.fbme.lib.iec61499.parser.ConverterArguments;
import org.fbme.lib.iec61499.parser.FBNetworkConverter;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class CompositeFbTypeNxtImporter extends CompositeFBTypeConverter {

    public CompositeFbTypeNxtImporter(ConverterArguments arguments) {
        super(arguments);
    }

    @Override
    @NotNull
    protected FBNetworkConverter createFBNetworkConverter(CompositeFBTypeDeclaration fbtd) {
        return new FBNetworkNxtImporter(with(Objects.requireNonNull(getElement()).getChild("FBNetwork")), fbtd.getNetwork());
    }
}

File: code/nxt-integration/src/main/kotlin/org/fbme/integration/nxt/importer/CompositeFbTypeNxtImporter.kt
package org.fbme.integration.nxt.importer

import org.fbme.lib.iec61499.declarations.CompositeFBTypeDeclaration
import org.fbme.lib.iec61499.parser.CompositeFBTypeConverter
import org.fbme.lib.iec61499.parser.ConverterArguments
import org.fbme.lib.iec61499.parser.FBNetworkConverter
import java.util.*

class CompositeFbTypeNxtImporter(arguments: ConverterArguments) : CompositeFBTypeConverter(arguments) {
    override fun createFBNetworkConverter(fbtd: CompositeFBTypeDeclaration): FBNetworkConverter {
        val element = checkNotNull(element)
        return FBNetworkNxtImporter(with(element.getChild("FBNetwork")), fbtd.network)
    }
}